### DISCUSSION TITLE
Is there a reason that both `AsRef` and `AsMut` are gated behind only 1 feature called `as_ref` but `Deref` and `DerefMut` require separate features called `deref` and `deref_mut` respectively?

### DISCUSSION THREAD

**Discussion by KSXGitHub** [JelteF/derive_more#422] [Q&A]
This asymmetry is pretty confusing.

**Comment by tyranron** [comment]
@KSXGitHub no, it's just `Deref` and `DerefMut` haven't been refactored yet, while `AsRef`/`AsMut` have been and are represented by the same actual code.
